deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/circuitbreaker/CircuitBreakerConfigurationParser.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkiverse.openapi.generator.deployment.circuitbreaker;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+import io.quarkiverse.openapi.generator.deployment.circuitbreaker.CircuitBreakerConfiguration.CircuitBreakerClassConfiguration;
+
+public final class CircuitBreakerConfigurationParser {
+
+    private final UnaryOperator<String> nameToValuePropertyMapper;
+
+    public CircuitBreakerConfigurationParser(UnaryOperator<String> nameToValuePropertyMapper) {
+        this.nameToValuePropertyMapper = nameToValuePropertyMapper;
+    }
+
+    public CircuitBreakerConfiguration parse(Collection<String> propertyNames) {
+        return new CircuitBreakerConfiguration(readClasses(propertyNames));
+    }
+
+    private List<CircuitBreakerClassConfiguration> readClasses(Collection<String> propertyNames) {
+        List<String> filteredPropertyNames = filterPropertyNames(propertyNames).stream()
+                .filter(property -> nameToValuePropertyMapper.apply(property).equals("true"))
+                .collect(Collectors.toList());
+
+        Set<String> classNames = filteredPropertyNames.stream()
+                .map(this::getClassName)
+                .collect(Collectors.toSet());
+
+        return classNames.stream()
```

Comment:
When using config, You can rewrite the whole method this way (still using functional approach)
```
 List<String> filteredPropertyNames = StreamSupport.stream(config.getPropertyNames().spliterator()).filter(property->config.getConfigValue(property,Boolean.class).collect(Collectors.toList());

return filteredPropertyNames.stream().map(this::getClassName).collect(Collectors.toMap(c ->c, c->getMethodNames(c,filteredPropertyNames));
```